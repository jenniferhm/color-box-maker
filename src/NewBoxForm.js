import React from 'react';


class NewBoxForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = { height: "", width: "", color: "" };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(evt) {
    evt.preventDefault();
    this.props.addBox(this.state);
    this.setState({ height: "", width: "", color: "" });
  }

  handleChange(evt) {
    this.setState({ [evt.target.name]: evt.target.value})
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <label htmlFor="height">Height:</label>
        <input id="height" name="height" value={this.state.height}
          onChange={this.handleChange} />

        <label htmlFor="width">Width:</label>
        <input id="width"  name="width" value={this.state.width}
          onChange={this.handleChange} />

        <label htmlFor="color">Color:</label>
        <input id="color" name="color" value={this.state.color}
          onChange={this.handleChange} />

        <button>Add a new box!</button>
      </form>
    )
  }
}


export default NewBoxForm;