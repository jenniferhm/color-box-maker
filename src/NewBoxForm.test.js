import React from "react";
import { shallow, mount} from "enzyme";
import toJson from "enzyme-to-json";
import NewBoxForm from './NewBoxForm';

it("renders without crashing", function() {
  shallow(<NewBoxForm />);
});

it("matches snapshot", function() {
  let wrapper = shallow(<NewBoxForm />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
})

it("allows for change of new Box", function() {
  let wrapper = mount(<NewBoxForm />);
  const heightInput = wrapper.find("#height");
  heightInput.instance().value = "200px"
  heightInput.simulate("change");

  expect(wrapper.state().height).toEqual("200px");

  const widthInput = wrapper.find("#width");
  widthInput.instance().value = "200px"
  widthInput.simulate("change");

  expect(wrapper.state().width).toEqual("200px");

  const colorInput = wrapper.find("#color");
  colorInput.instance().value = "blue"
  colorInput.simulate("change");

  expect(wrapper.state().color).toEqual("blue");
});

it("runs a mocked addBox function on submit", function() {
  const submitFn = jest.fn();
  let wrapper = mount(<NewBoxForm addBox={submitFn} />);
  const form = wrapper.find("form");

  form.simulate("submit");

  expect(submitFn).toHaveBeenCalled();
});

it("resets state on submit", function() {
  const submitFn = jest.fn();
  let wrapper = mount(<NewBoxForm addBox={submitFn} />);

  const heightInput = wrapper.find("#height");
  heightInput.instance().value = "200px"
  heightInput.simulate("change");

  expect(wrapper.state().height).toEqual("200px");

  const form = wrapper.find("form");
  form.simulate("submit");

  expect(wrapper.state()).toEqual({
    height: "",
    width: "",
    color: ""
  });
});