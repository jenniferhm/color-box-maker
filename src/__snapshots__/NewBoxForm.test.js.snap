// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="height"
  >
    Height:
  </label>
  <input
    id="height"
    name="height"
    onChange={[Function]}
    value=""
  />
  <label
    htmlFor="width"
  >
    Width:
  </label>
  <input
    id="width"
    name="width"
    onChange={[Function]}
    value=""
  />
  <label
    htmlFor="color"
  >
    Color:
  </label>
  <input
    id="color"
    name="color"
    onChange={[Function]}
    value=""
  />
  <button>
    Add a new box!
  </button>
</form>
`;
